code/modules/paperwork/pen/pen.dm

```diff
@@ -1,49 +1,34 @@
 /obj/item/pen
-	desc = "It's a normal black ink pen."
-	name = "pen"
-	icon = 'icons/obj/items/pens/pen.dmi'
-	icon_state = ICON_STATE_WORLD
-	slot_flags = SLOT_LOWER_BODY | SLOT_EARS
-	throwforce = 0
-	w_class = ITEM_SIZE_TINY
-	throw_speed = 7
-	throw_range = 15
-	material = /decl/material/solid/plastic
-
-	var/colour = "black"	//what colour the ink is!
-	var/color_description = "black ink"
-	var/active = TRUE
-	var/iscrayon = FALSE
-	var/isfancy = FALSE
+	name                   = "pen"
+	desc                   = ""
+	icon                   = 'icons/obj/items/pens/pen.dmi'
+	icon_state             = ICON_STATE_WORLD
+	slot_flags             = SLOT_LOWER_BODY | SLOT_EARS
+	w_class                = ITEM_SIZE_TINY
+	throwforce             = 0
+	throw_speed            = 7
+	throw_range            = 15
+	material               = /decl/material/solid/plastic
+	var/pen_flag           = PEN_FLAG_ACTIVE                     //Properties/state of the pen used.
+	var/stroke_colour      = "black"                             //What colour the ink is! Can be hexadecimal colour or a colour name string.
+	var/stroke_colour_name = "black"                             //Human readable name of the stroke colour. Used in text strings, and to identify the nearest colour to the stroke colour.
+	var/medium_name        = "ink"                               //Whatever the pen uses to leave its mark. Used in text strings.
+	var/max_uses           = -1                                  //-1 for unlimited uses.
+	var/pen_quality        = TOOL_QUALITY_DEFAULT                //What will be set as tool quality for the pen
 
 /obj/item/pen/Initialize(ml, material_key)
 	. = ..()
-	set_extension(src, /datum/extension/tool, list(TOOL_DRILL = TOOL_QUALITY_WORST))
-
-/obj/item/pen/blue
-	name = "blue pen"
-	desc = "It's a normal blue ink pen."
-	icon = 'icons/obj/items/pens/pen_blue.dmi'
-	colour = "blue"
-	color_description = "blue ink"
-
-/obj/item/pen/red
-	name = "red pen"
-	desc = "It's a normal red ink pen."
-	icon = 'icons/obj/items/pens/pen_red.dmi'
-	colour = "red"
-	color_description = "red ink"
-
-/obj/item/pen/green
-	name = "green pen"
-	desc = "It's a normal green ink pen."
-	icon = 'icons/obj/items/pens/pen_green.dmi'
-	colour = "green"
-
-/obj/item/pen/invisible
-	desc = "It's an invisble pen marker."
-	colour = "white"
-	color_description = "transluscent ink"
+	set_extension(src, /datum/extension/tool,
+		list(
+				TOOL_DRILL = TOOL_QUALITY_WORST,
```

Comment:
Double tabbing here? Not a blocking issue, just looks a bit weird.

Reply:
True, I'll fix that.